src/lib.rs

```diff
@@ -945,6 +945,40 @@ safety_comment! {
     unsafe_impl!(T: ?Sized + Unaligned => Unaligned for ManuallyDrop<T>);
     assert_unaligned!(ManuallyDrop<()>, ManuallyDrop<u8>);
 }
+safety_comment! {
+    /// SAFETY:
+    /// The all-zeroes const and mut raw pointers are valid, and it is sound to
+    /// materialize them from nothing. The existence of `ptr::null` [1] and
```

Comment:
Is it clearly documented anywhere that `0` is a valid value for the null pointer? Infamously, the C standard does not require that null pointers have an all-zero bit-pattern, only that they compare as equal to `0`.